src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -1099,7 +1099,7 @@ private bool ShouldSkipCommandResolutionForConstrainedLanguage(CommandInfo? resu
                 // Relative Path:       ".\command.exe"
                 // Home Path:           "~\command.exe"
                 // Drive Relative Path: "\Users\User\AppData\Local\Temp\command.exe"
-                if (_commandName[0] == '.' || _commandName[0] == '~' || _commandName[0] == '\\')
+                if (!string.IsNullOrEmpty(_commandName) && (_commandName[0] == '.' || _commandName[0] == '~' || _commandName[0] == '\\'))
```

Comment:
`_commandName` is defined as non-nullable so we can do not check null.
Also it is on a hot path and we could optimize the code a bit. I suggest add `&& _commandName.Length != 0` in `if` above, define `char firstChar = _commandName[0]` and use the `firstChar` in the line.

Reply:
@iSazonov just to confirm. Is this along the lines of what you're suggesting? 
`                char firstChar = _commandName[0];`
                `if (_commandName.Length != 0 && (firstChar == '.' || firstChar == '~' || firstChar == '\\'))`

I agree it would be more optimized but the issue of this PR (`Get-Command ' '` returns the wrong exception) wouldn't be fixed. Sorry I may be misunderstanding. I'm still learning.

Reply:
> (!string.IsNullOrEmpty(_commandName)

You added the check to fix the issue. But `_commandName` is annotated as non-nullable and we can exclude null check and have `_commandName.Length != 0`. And the check should be in previous `if` condition (before `char firstChar = _commandName[0]`).